main.rb

```diff
@@ -1,5 +1,4 @@
 # rubocop:disable Metrics/CyclomaticComplexity
-# rubocop:disable Metrics/PerceivedComplexity
 # rubocop:disable Metrics/MethodLength
```

Comment:
## [Optional]

If you follow the single responsibility principle, you wouldn't have disabled this linter config 🙂 - kindly read more [here](https://thoughtbot.com/blog/sandi-metz-rules-for-developers)

Reply:
@bellom I didn't find any valid real-world reason in the context of this project in the provided link for the main menu options. In real-world applications, this data would usually come as a JSON or YAML object which could be directly mapped using `loops` without the `switch cases`. In that case, method lengths will automatically reduce, but here the main has to take inputs and pass them forward.

**N.B: Now how can tnat be shortened for different `cases` I have no valid idea. If you could guide me that would be great. I can meet with you over a zoom**

**In my advocate:** When you check the `main` function, it's having only one responsibility as project required - 
 > taking user inputs & passing to a dependent object (`app` here)

- It's open and closed, just adding an extra 1 case will take care of another functionality
- The Inherited classes parent and child classes are substitutable (**Liskov substitution**)
- Similar types of classes are having different responsibilities but similar implementations using interfaces - (**Interface segregation**)
- In the models, the **Dependency Inversion** followed while using **decorator pattern**